packages/strapi-helper-plugin/lib/src/utils/request.js

```diff
@@ -82,11 +82,20 @@ function serverRestartWatcher(response) {
    // Set headers
    options.headers = Object.assign({
```

Comment:
The order of the objects assigned is not correct to ensure (100%) that the `X-Forwarded-Host` will be set to the right value.

```js
options.headers = Object.assign({
  'Content-Type': 'application/json'
}, options.headers, {
  'X-Forwarded-Host': 'strapi',
});
```